spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -314,6 +318,10 @@ private boolean canDeleteUsingMetadata(Expression deleteExpr) {
     }
   }
 
+  private boolean isCaseSensitive() {
+    return Boolean.parseBoolean(sparkSession().conf().get("spark.sql.caseSensitive"));
```

Comment:
[minor] should we make this instance variable ?

Reply:
I also thought about that but it would mean initializing the `SparkSession` var in the constructor. I don't recall why we decided to avoid that initially but I'd probably keep the current logic to prevent surprises.

Reply:
Changed this bit back to have a local variable.